.github/workflows/build.yml

```diff
@@ -0,0 +1,29 @@
+name: Build
+
+on:
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    permissions:
+      packages: read
+
+    container:
+      image: ghcr.io/access-nri/oasis3-mct-build-image:pr-10
+      credentials:
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+
+    steps:
+    - name: Build oasis3-mct via spack
+      run: spack -v install --only package access.nri.oasis3-mct@$GITHUB_REF_NAME
+      if: github.event_name != 'pull_request'
+
+    # Workaround for this issue: https://github.com/github/docs/issues/15319
```

Comment:
Yeah, I had a one-liner solution to this in bash but decided against using it, imo it's preferable not to put logic in the run command.

```
spack -v install --only package access.nri.oasis3-mct@`if [ -z ${GITHUB_HEAD_REF+x} ]; then echo $GITHUB_HEAD_REF; else echo $GITHUB_REF_NAME; fi`
```

CI_ACTION_REF_NAME from that action looks like it might do what we want. Something to test once the pipeline is working.